runtime/caml/platform.h

```diff
@@ -28,18 +28,23 @@
 #define MAP_ANONYMOUS MAP_ANON
 #endif
 
-/* Loads and stores with acquire and release semantics respectively */
+/* Hint for busy-waiting loops */
 
 Caml_inline void cpu_relax() {
+#ifdef __GNUC__
 #if defined(__x86_64__) || defined(__i386__)
   asm volatile("pause" ::: "memory");
 #elif defined(__aarch64__)
   asm volatile ("yield" ::: "memory");
 #else
-  #warning "cpu_relax() undefined for this architecture!"
+  /* Just a compiler barrier */
+  asm volatile ("" ::: "memory");
+#endif
```

Comment:
Would it be possible to keep an `#warning` here, possibly as:
```suggestion
#elif defined(_MSC_VER)
  #warning "cpu_relax() not yet defined for this architecture!"
#endif
```
For MSVC, I think it's a matter of including `<intrin.h>` somewhere and then something like:
```c
  _mm_pause();
  _ReadWriteBarrier();
```

Reply:
To me, MSVC support is quite far in the future (unless they start supporting C11 quickly), so I'll keep the suggestion in mind, but not in the code.

Reply:
Recent versions alreasy support C11 according to https://devblogs.microsoft.com/cppblog/c11-and-c17-standard-support-arriving-in-msvc/

Reply:
I meant C11 atomics.  From the blog post:

> While there is currently no support for any C11 optional features, we are committed to providing the most impactful optional features in future releases. Atomic and threading support are on our roadmap.